### PR TITLE
New C++ API: Term: Add missing checks for null.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/api/cvc4cpp.cpp [hunk 1]
```diff
@@ -1080,6 +1081,7 @@ Term Term::notTerm() const
 
 Term Term::andTerm(const Term& t) const
 {
+  CVC4_API_CHECK_NOT_NULL;
```
Comment (4tXJ7f) [inline src/api/cvc4cpp.cpp]: Maybe we should also check `t` for being null? What do you think?

Comment (aniemetz) [reply src/api/cvc4cpp.cpp]: Good point, will add.

Comment (aniemetz) [reply src/api/cvc4cpp.cpp]: Done, PTAL

File: test/unit/api/term_black.h [hunk 2]
```diff
@@ -182,6 +183,7 @@ void TermBlack::testAndTerm()
   Sort funSort2 = d_solver.mkFunctionSort(intSort, boolSort);
 
   Term b = d_solver.mkTrue();
+  TS_ASSERT_THROWS(Term().andTerm(b), CVC4ApiException&);
```
Comment (4tXJ7f) [inline test/unit/api/term_black.h]:
```suggestion
  TS_ASSERT_THROWS(Term().andTerm(b), CVC4ApiException&);
  TS_ASSERT_THROWS(b.andTerm(Term()), CVC4ApiException&);  
```

File: test/unit/api/term_black.h [hunk 3]
```diff
@@ -246,6 +248,7 @@ void TermBlack::testOrTerm()
   Sort funSort2 = d_solver.mkFunctionSort(intSort, boolSort);
 
   Term b = d_solver.mkTrue();
+  TS_ASSERT_THROWS(Term().orTerm(b), CVC4ApiException&);
```
Comment (4tXJ7f) [inline test/unit/api/term_black.h]:
```suggestion
  TS_ASSERT_THROWS(Term().orTerm(b), CVC4ApiException&);
  TS_ASSERT_THROWS(b.orTerm(Term()), CVC4ApiException&);
```

File: test/unit/api/term_black.h [hunk 4]
```diff
@@ -310,6 +313,7 @@ void TermBlack::testXorTerm()
   Sort funSort2 = d_solver.mkFunctionSort(intSort, boolSort);
 
   Term b = d_solver.mkTrue();
+  TS_ASSERT_THROWS(Term().xorTerm(b), CVC4ApiException&);
```
Comment (4tXJ7f) [inline test/unit/api/term_black.h]:
```suggestion
  TS_ASSERT_THROWS(Term().xorTerm(b), CVC4ApiException&);
  TS_ASSERT_THROWS(b.xorTerm(Term()), CVC4ApiException&);
```

File: test/unit/api/term_black.h [hunk 5]
```diff
@@ -374,6 +378,7 @@ void TermBlack::testEqTerm()
   Sort funSort2 = d_solver.mkFunctionSort(intSort, boolSort);
 
   Term b = d_solver.mkTrue();
+  TS_ASSERT_THROWS(Term().eqTerm(b), CVC4ApiException&);
```
Comment (4tXJ7f) [inline test/unit/api/term_black.h]:
```suggestion
  TS_ASSERT_THROWS(Term().eqTerm(b), CVC4ApiException&);
  TS_ASSERT_THROWS(b.eqTerm(Term()), CVC4ApiException&);
```

File: test/unit/api/term_black.h [hunk 6]
```diff
@@ -438,6 +443,7 @@ void TermBlack::testImpTerm()
   Sort funSort2 = d_solver.mkFunctionSort(intSort, boolSort);
 
   Term b = d_solver.mkTrue();
+  TS_ASSERT_THROWS(Term().impTerm(b), CVC4ApiException&);
```
Comment (4tXJ7f) [inline test/unit/api/term_black.h]:
```suggestion
  TS_ASSERT_THROWS(Term().impTerm(b), CVC4ApiException&);
  TS_ASSERT_THROWS(b.impTerm(Term()), CVC4ApiException&);  
```

File: test/unit/api/term_black.h [hunk 7]
```diff
@@ -502,6 +508,7 @@ void TermBlack::testIteTerm()
   Sort funSort2 = d_solver.mkFunctionSort(intSort, boolSort);
 
   Term b = d_solver.mkTrue();
+  TS_ASSERT_THROWS(Term().iteTerm(b, b), CVC4ApiException&);
```
Comment (4tXJ7f) [inline test/unit/api/term_black.h]:
```suggestion
  TS_ASSERT_THROWS(Term().iteTerm(b, b), CVC4ApiException&);
  TS_ASSERT_THROWS(b.iteTerm(Term(), b), CVC4ApiException&);
  TS_ASSERT_THROWS(b.iteTerm(b, Term()), CVC4ApiException&);    
```

**Comment by 4tXJ7f** [comment]
Marking as "do not merge" to give a chance to apply the suggestions.

**Pull request merged**